### PR TITLE
add overload tests and tool tests

### PR DESCRIPTION
**Pull request by sophia-ramsey**
environment variable currently named `azure_ai_projects_connection_string_agents_tests `-- may need to update for consistency

still TODO: 
- streaming tests
- more tool tests
- testing for tools with functions of different input parameter types

### CODE REVIEW

File: sdk/ai/azure-ai-projects/tests/agents/test_agents_client.py [hunk 1]
```diff
@@ -45,13 +46,13 @@
 
 agentClientPreparer = functools.partial(
     EnvironmentVariableLoader,
-    "azure_ai_project",
-    project_connection_string_agents_tests="https://foo.bar.some-domain.ms;00000000-0000-0000-0000-000000000000;rg-resour-cegr-oupfoo1;abcd-abcdabcdabcda-abcdefghijklm",
+    "azure_ai_projects",
+    azure_ai_projects_connection_string_agents_tests="foo.bar.some-domain.ms;00000000-0000-0000-0000-000000000000;rg-resour-cegr-oupfoo1;abcd-abcdabcdabcda-abcdefghijklm",
```
Comment (dargilco) [inline sdk/ai/azure-ai-projects/tests/agents/test_agents_client.py:50]: Yes, per your comment, please update this to `azure_ai_projects_agents_tests_project_connection_string` (corresponding to env variable `AZURE_AI_PROJECTS_AGENTS_TESTS_PROJECT_CONNECTION_STRING`)

File: sdk/ai/azure-ai-projects/tests/agents/test_agents_tools.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+    logger.setLevel(logging.DEBUG)  # INFO or DEBUG
+
+    # Configure a console output
+    handler = logging.StreamHandler(stream=sys.stdout)
+    logger.addHandler(handler)
+
+agentClientPreparer = functools.partial(
+    EnvironmentVariableLoader,
+    "azure_ai_projects",
+    azure_ai_projects_connection_string_agents_tests="foo.bar.some-domain.ms;00000000-0000-0000-0000-000000000000;rg-resour-cegr-oupfoo1;abcd-abcdabcdabcda-abcdefghijklm",
```
Comment (dargilco) [inline sdk/ai/azure-ai-projects/tests/agents/test_agents_tools.py:46]: Same as above

**Comment by dargilco** [comment]
Add to the TODO list: tests for async client

File: sdk/ai/azure-ai-projects/tests/agents/test_agents_tools.py [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+    # TODO test upload file
+
+    # test assistant creation with file_search and add_vector_store
+    @agentClientPreparer()
+    @recorded_by_proxy
+    def test_file_search_add_vector_store(self, **kwargs):
+        # Create client
+        client = self.create_client(**kwargs)
+        assert isinstance(client, AIProjectClient)
```
Comment (dargilco) [inline sdk/ai/azure-ai-projects/tests/agents/test_agents_tools.py:162]: Please also have some (or all) of the tests use the "with ... as ..." pattern, to test calls to `__enter__` and `__exit__` methods. So:
`with self.create_client() as client:`

In this case you don't need an explicit call to `client.close()` at the end

File: sdk/ai/azure-ai-projects/tests/agents/test_agents_tools.py [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+from azure.core.pipeline.transport import RequestsTransport
+from devtools_testutils import AzureRecordedTestCase, EnvironmentVariableLoader, recorded_by_proxy
+from azure.core.exceptions import AzureError, ServiceRequestError, HttpResponseError
+from azure.identity import DefaultAzureCredential
+
+# TODO clean this up / get rid of anything not in use
+
+'''
+issues I've noticed with the code: 
+    delete_thread(thread.id) fails
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/tests/agents/test_agents_tools.py:24]: Please create issue/task on backlog for this. Is it consistently failing?

File: sdk/ai/azure-ai-projects/tests/agents/test_agents_tools.py [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+    cancel_thread(thread.id) expires/times out occasionally
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/tests/agents/test_agents_tools.py:25]: Maybe good to have issue about this also, is there a reason information in cancellation?

File: sdk/ai/azure-ai-projects/tests/agents/test_agents_tools.py [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+import sys
+
+from azure.ai.projects import AIProjectClient
+from azure.ai.projects.models import FunctionTool, CodeInterpreterTool, FileSearchTool, ToolSet, VectorStore
+from azure.core.pipeline.transport import RequestsTransport
+from devtools_testutils import AzureRecordedTestCase, EnvironmentVariableLoader, recorded_by_proxy
+from azure.core.exceptions import AzureError, ServiceRequestError, HttpResponseError
+from azure.identity import DefaultAzureCredential
+
+# TODO clean this up / get rid of anything not in use
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/tests/agents/test_agents_tools.py:20]: Please make the clean-up for comments below after creating tasks

File: sdk/ai/azure-ai-projects/tests/agents/test_agents_client.py [hunk 7]
```diff
@@ -86,23 +87,23 @@ def fetch_current_datetime_recordings():
...
 
 # Statically defined user functions for fast reference
-user_functions_recording = {"fetch_current_datetime": fetch_current_datetime_recordings}
-user_functions_live = {"fetch_current_datetime": fetch_current_datetime_live}
+user_functions_recording = [fetch_current_datetime_recordings]
+user_functions_live = [fetch_current_datetime_live]
 
 
 # The test class name needs to start with "Test" to get collected by pytest
 class TestagentClient(AzureRecordedTestCase):
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/tests/agents/test_agents_client.py:95]: small nit, class names are normally Pascal case -> TestAgentClient

File: sdk/ai/azure-ai-projects/tests/agents/test_agents_client.py [hunk 8]
```diff
@@ -216,6 +272,103 @@ def test_update_agent(self, **kwargs):
...
+        agent = client.agents.update_agent(agent.id, name="my-agent2")
+        assert agent.name
+        assert agent.name == "my-agent2"
+
+        # delete agent and close client
+        client.agents.delete_agent(agent.id)
+        print("Deleted agent")
+        client.close()
+
+    # NOTE update_agent with overloads isn't working
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/tests/agents/test_agents_client.py:305]: please create issues on tests that are not working to backlog

File: sdk/ai/azure-ai-projects/tests/agents/test_agents_client.py [hunk 9]
```diff
@@ -269,7 +422,7 @@ def test_create_thread(self, **kwargs):
 
         # create thread
         thread = client.agents.create_thread()
-        # assert isinstance(thread, agentThread) TODO finish this ! need to import agentThread from _models
```
Comment (jhakulin) [inline sdk/ai/azure-ai-projects/tests/agents/test_agents_client.py:272]: is this TODO still valid?

File: sdk/ai/azure-ai-projects/tests/agents/test_agents_client.py (hunk 8 above)
Comment (sophia-ramsey) [reply sdk/ai/azure-ai-projects/tests/agents/test_agents_client.py:305]: sorry, I meant to delete that note

**Pull request closed without merging**